libqtile/widget/keyboardlayout.py

```diff
@@ -99,15 +99,20 @@ def set_keyboard(self, layout: str, options: Optional[str]) -> None:
 
 
 class _WaylandLayoutBackend(_BaseLayoutBackend):
-    def __init__(self, qtile: Qtile):
+    def __init__(self, qtile: Qtile) -> None:
         self.set_keymap = qtile.core.set_keymap  # type: ignore
         self._layout: str = ""
 
     def get_keyboard(self) -> str:
         return self._layout
 
     def set_keyboard(self, layout: str, options: Optional[str]) -> None:
-        self.set_keymap(layout, options)
+        maybe_variant: Optional[str] = None
+        if " " in layout:
+            layout_name, _, maybe_variant = layout.partition(" ")
```

Comment:
Why `partition` over `split`?

Reply:
Right, initially I had always done `layout.partition(" ")`, which is nice in that it always returns a 3-tuple, unlike split which would just return the input as a 1-tuple is the separator character isn't present, but then I had to transform the empty string to `None` to pass to `set_keymap`, which the type checker doesn't like, and you can't annotate a single part of the tuple when descructuring, like `layout_name, _, (maybe_variant: Optional[str]) = layout.partition(" ")`. Ultimately this felt like the better logical layout, but now that we know `" "` is in the layout, using `split` makes sense here.

Reply:
Fair - when reading this my next question was going to be "ooh how are we going to limit the splits to 2 items?". I see now there is an argument for that.